products/bash/mq/Readme.md

```diff
@@ -0,0 +1,216 @@
+# Using TLS to communicate between ACE and MQ
+
+## Setting up MQ to use TLS
+
+### Create Queues 
+- Queues need to be configured to use:
+ 
+ - SSL
+ - Assigning user to Queues
+ - Assigning permission to the User
+
+ ```
+    DEFINE QLOCAL('AccidentIn') DEFPSIST(YES) \n\
+    DEFINE QLOCAL('AccidentOut') DEFPSIST(YES) \n\
+    DEFINE QLOCAL('BumperIn') DEFPSIST(YES) \n\
+    DEFINE QLOCAL('BumperOut') DEFPSIST(YES) \n\
+    DEFINE QLOCAL('CrumpledIn') DEFPSIST(YES) \n\
+    DEFINE QLOCAL('CrumpledOut') DEFPSIST(YES) \n\
+    DEFINE CHANNEL(ACE_SVRCONN) CHLTYPE(SVRCONN) TRPTYPE(TCP) SSLCAUTH(REQUIRED) SSLCIPH('ECDHE_RSA_AES_128_CBC_SHA256') \n\
+    ALTER AUTHINFO(SYSTEM.DEFAULT.AUTHINFO.IDPWOS) AUTHTYPE(IDPWOS) ADOPTCTX(YES) CHCKCLNT(OPTIONAL) CHCKLOCL(OPTIONAL) AUTHENMD(OS) \n\
+    SET CHLAUTH('ACE_SVRCONN') TYPE(SSLPEERMAP) SSLPEER('CN=application1,OU=app team1') USERSRC(MAP) MCAUSER('mqm') ACTION(ADD) \n\
+    REFRESH SECURITY TYPE(CONNAUTH) \n\
+    SET AUTHREC PRINCIPAL('mqm') OBJTYPE(QMGR) AUTHADD(CONNECT,INQ) \n\
+    SET AUTHREC PROFILE('AccidentIn') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+    SET AUTHREC PROFILE('AccidentOut') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+    SET AUTHREC PROFILE('BumperIn') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+    SET AUTHREC PROFILE('BumperOut') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+    SET AUTHREC PROFILE('CrumpledIn') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+    SET AUTHREC PROFILE('CrumpledOut') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+```
+
+- Setup certs,keys,keystores,truststores for MQ
+    
+    We have used this [products\bash\mq\createcerts\generate-test-cert.sh] to create the certs.
+    When creating application.kdb we need to add the `aceclient` label so that ACE Integration Servers can recognize it
+
+    I used the following command to add the label:
+
+    `runmqckm -cert -import -file application.p12 -pw password -type pkcs12 -target application.kdb -target_pw password -target_type cms -label "1" -new_label aceclient`
```

Comment:
Could we add a little bit of info as to what runmqckm is please? I.e. if I were wanting to use it, how would I get it, or who would I contact for more info?